### PR TITLE
AWS neocna resources

### PR DESCRIPTION
**Pull request by sawyerawesome**
## Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by sawyerawesome** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "pccnabackend",
    "pr-id": "201",
    "commit-sha": "08f2d7b78ccf1c0c6f7e7eb2f723620a1e1a68fb",
    "pipeline": "master",
    "org": "prismacloud/pcn"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "897",
    "commit-sha": "a6b6fb172119751282fe19637fbf1e64e486822c",
    "pipeline": "master",
    "org": "aporeto-inc"
  }
]
```

**Comment by kcarrasco-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "neocna-aws-skeleton",
    "component": "pccnabackend",
    "pr-id": "201",
    "commit-sha": "ef1474fc1b503e93d7b7b433a6aaeddb5c59f860",
    "pipeline": "master",
    "org": "prismacloud/pcn"
  },
  {
    "project": "neocna-aws-skeleton",
    "component": "gaia",
    "pr-id": "897",
    "commit-sha": "a6b6fb172119751282fe19637fbf1e64e486822c",
    "pipeline": "master",
    "org": "aporeto-inc"
  }
]
```

**Review by kcarrasco-panw — APPROVED** [review]

**Comment by kcarrasco-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "neocna-aws-skeleton",
    "component": "gaia",
    "pr-id": "897",
    "commit-sha": "a6b6fb172119751282fe19637fbf1e64e486822c",
    "pipeline": "master",
    "org": "aporeto-inc"
  },
  {
    "project": "neocna-aws-skeleton",
    "component": "pccnabackend",
    "pr-id": "201",
    "commit-sha": "58d4688dc55952579e86fa1604d212e8fffa0289",
    "pipeline": "master",
    "org": "prismacloud/pcn"
  }
]
```

**Comment by kcarrasco-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "neocna-aws-skeleton",
    "component": "gaia",
    "pr-id": "897",
    "commit-sha": "a6b6fb172119751282fe19637fbf1e64e486822c",
    "pipeline": "master",
    "org": "aporeto-inc"
  },
  {
    "project": "neocna-aws-skeleton",
    "component": "pccnabackend",
    "pr-id": "201",
    "commit-sha": "a35b001060f66cc2a411afc9d42ef64c724d9fce",
    "pipeline": "master",
    "org": "prismacloud/pcn"
  }
]
```

**Comment by kcarrasco-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "neocna-aws-skeleton",
    "component": "gaia",
    "pr-id": "897",
    "commit-sha": "a6b6fb172119751282fe19637fbf1e64e486822c",
    "pipeline": "master",
    "org": "aporeto-inc"
  },
  {
    "project": "neocna-aws-skeleton",
    "component": "pccnabackend",
    "pr-id": "201",
    "commit-sha": "27c268acb061aab2d15009853b30b4a680fe5966",
    "pipeline": "master",
    "org": "prismacloud/pcn"
  }
]
```

**Comment by kcarrasco-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "neocna-aws-skeleton",
    "component": "gaia",
    "pr-id": "897",
    "commit-sha": "9723f71eb9b5c1ad26c36f4f3b817a6fef36e438",
    "pipeline": "master",
    "org": "aporeto-inc"
  },
  {
    "project": "neocna-aws-skeleton",
    "component": "pccnabackend",
    "pr-id": "201",
    "commit-sha": "27c268acb061aab2d15009853b30b4a680fe5966",
    "pipeline": "master",
    "org": "prismacloud/pcn"
  }
]
```

File: specs/@awsresourceattrs.abs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    example_value: Instance
+
+  - name: name
+    description: The name of the resource.
+    type: string
+    exposed: true
+    stored: true
+    example_value: abc
+
+  - name: prismaAPIID
```
Comment (malkhamis-panw) [inline specs/@awsresourceattrs.abs:66]: we don't need this

File: specs/@awsresourceattrs.abs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      {
+        "id": "0000000000000000000",
+        "kind": "compute#instance"
+        ...
+      }
+
+  - name: denormedFields
+    description: |-
+      Contextual values that can be used to narrow searching of resources if the
+      rrn or resourceID are not known. For instance, it could be used to store
```
Comment (malkhamis-panw) [inline specs/@awsresourceattrs.abs:35]: `arn` instead of `rrn`

File: specs/@awsresourceattrs.abs [hunk 3]
```diff
@@ -0,0 +1,112 @@
+# Attributes
+attributes:
+  v1:
+  - name: accountID
```
Comment (malkhamis-panw) [inline specs/@awsresourceattrs.abs:4]: also include these flags
```
    read_only: true
    autogenerated: true
```

File: specs/@awsresourceattrs.abs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    description: The ID of the account the resource belongs to in AWS.
+    type: string
+    exposed: true
+    stored: true
+    omit_empty: true
+
+  - name: arn
```
Comment (malkhamis-panw) [inline specs/@awsresourceattrs.abs:11]: also include these flags
```
    read_only: true
    autogenerated: true
```

File: specs/@awsresourceattrs.abs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    stored: true
+    required: true
+    example_value: |-
+      {
+        "id": "0000000000000000000",
+        "kind": "compute#instance"
+        ...
+      }
+
+  - name: denormedFields
```
Comment (malkhamis-panw) [inline specs/@awsresourceattrs.abs:32]: also include these flags
```
    read_only: true
    autogenerated: true
```

File: specs/@awsresourceattrs.abs [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    description: |-
+      Contextual values that can be used to narrow searching of resources if the
+      rrn or resourceID are not known. For instance, it could be used to store
+      a resource's Subnet or VPC ID.
+    type: list
+    exposed: true
+    subtype: string
+    stored: true
+
+  - name: kind
```
Comment (malkhamis-panw) [inline specs/@awsresourceattrs.abs:42]: also include these flags
```
    read_only: true
    autogenerated: true
```

File: specs/@awsresourceattrs.abs [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    - NetworkInterface
+    - VPC
+    - Subnet
+    - NetworkACL
+    - InternetGateway
+    - SecurityGroup
+    default_value: Pending
+    example_value: Instance
+
+  - name: name
```
Comment (malkhamis-panw) [inline specs/@awsresourceattrs.abs:59]: also include these flags
```
    read_only: true
    autogenerated: true
```

File: specs/@awsresourceattrs.abs [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+  - name: prismaRegion
+    description: |-
+      The region this resource exists in according to PrismaCloud. Note that the
+      resource may exists in a different region as described by AWS.
+    type: string
+    exposed: true
+    stored: true
+    example_value: us-east-1
+
+  - name: resourceID
```
Comment (malkhamis-panw) [inline specs/@awsresourceattrs.abs:88]: also include these flags
```
    read_only: true
    autogenerated: true
```

File: specs/@awsresourceattrs.abs [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+    description: A resource ID that will mainly be used in RQL queries.
+    type: string
+    exposed: true
+    stored: true
+    example_value: i-092bf3d4b2422d8a6
+
+  - name: service
```
Comment (malkhamis-panw) [inline specs/@awsresourceattrs.abs:95]: also include these flags
```
    read_only: true
    autogenerated: true
```

**Comment by kcarrasco-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "neocna-aws-skeleton",
    "component": "gaia",
    "pr-id": "897",
    "commit-sha": "c1cfbb79211506486bf9598279d154bef9a92354",
    "pipeline": "master",
    "org": "aporeto-inc"
  },
  {
    "project": "neocna-aws-skeleton",
    "component": "pccnabackend",
    "pr-id": "201",
    "commit-sha": "821a34ddc659631b7e92938aaa7407515f8c7f3e",
    "pipeline": "master",
    "org": "prismacloud/pcn"
  }
]
```

**Review by kcarrasco-panw — APPROVED** [review]

**Review by malkhamis-panw — APPROVED** [review]

**Pull request merged**